primitives/account/src/logs.rs

```diff
@@ -121,6 +121,9 @@ pub enum Log {
         new_validator_address: Option<Address>,
     },
 
+    #[cfg_attr(feature = "serde-derive", serde(rename_all = "camelCase"))]
+    RetireValidator { validator_address: Address },
```

Comment:
```suggestion
    RetireValidator { validator_address: Address, newly_inactivated: bool },
```
@ii-cruz and I would suggest to add a bool to the log that is set to true if `retire` also inactivates the validator (when it is called from an active validator).
That way, when a block with a retire transaction is reverted, the listener can know which state the validator goes back into.

Reply:
When an active validator is retired, a `DeactivateValidator` log is emitted, too. Both are then emitted.

Reply:
https://github.com/nimiq/core-rs-albatross/pull/1547/files#diff-92583b503b91c6afdf466239768975d1071b25a871fdc50ab664652a6b686f6dR538

Reply:
What we don't log is `UnparkValidator` when we retire the validator. Even thought we unparked it [here](https://github.com/nimiq/core-rs-albatross/pull/1547/files#diff-92583b503b91c6afdf466239768975d1071b25a871fdc50ab664652a6b686f6dR544).

@paberr I already have the fix to your comment in my local. Let me know what you decided given the comments from @sisou 😃

Reply:
True, I missed the Deactivate log.

But then indeed we should do three things:

- We should also emit `RetireValidator` in the revert function.
- We should also emit `UnparkValidator` in the commit and revert for the retire (if it was parked).
- We don't need the boolean in `RetireValidator`, so remove it. Sorry about this 😅

Reply:
Regarding your bullet points: 
1. Emitting the log `RetireValidator` implies that the validator has changed to `retired` state. In the `revert` function, what we do is removing the state. Therefore, does it make sense to log `RetireValidator` when we change to `validator.retired = false`? I suggest to create a new log called `RevertRetireValidator`.
2. Not sure what you mean by the `commit` function
3. Ok

Reply:
1. No, they are using the same logs. But the fact that the log will be in an array called something with "reverted_logs", will make it clear that it was in fact reverted. See all the `revert_*` methods in `staking_contract/validator.rs`. So just add the same lines pushing the log into the logger in the `revert_retire_validator` method.
2. The "commit" method is the one applying the change so yes the `retire_validator` method.